huaweicloud/services/obs/resource_huaweicloud_obs_bucket_acl.go

```diff
@@ -0,0 +1,473 @@
+				Type:     schema.TypeString,
+				Required: true,
+				Description: `Specifies the domain id to authorize. The domain id cannot be the bucket owner, 
+and must be unique.`,
+			},
+			"access_to_bucket": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Description: `Specifies the access to bucket. Valid values are **READ** and **WRITE**.`,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+					ValidateFunc: validation.StringInSlice([]string{
+						"READ", "WRITE",
+					}, false),
+				},
+			},
+			"access_to_acl": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Description: `Specifies the access to acl. Valid values are **READ_ACP** and **WRITE_ACP**.`,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+					ValidateFunc: validation.StringInSlice([]string{
+						"READ_ACP", "WRITE_ACP",
+					}, false),
+				},
+			},
+		},
+	}
+	return &sc
+}
+
+func getPermissionTypesFromRawMap(rawMap map[string]interface{}) []obs.PermissionType {
+	var accesses []string
+	if accessArray, ok := rawMap["access_to_bucket"].([]interface{}); ok {
+		accesses = append(accesses, utils.ExpandToStringList(accessArray)...)
+	}
+	if accessArray, ok := rawMap["access_to_acl"].([]interface{}); ok {
+		accesses = append(accesses, utils.ExpandToStringList(accessArray)...)
+	}
+
+	var permissionTypes []obs.PermissionType
+	for _, access := range accesses {
+		switch access {
+		case "READ":
+			permissionTypes = append(permissionTypes, obs.PermissionRead)
+		case "WRITE":
+			permissionTypes = append(permissionTypes, obs.PermissionWrite)
+		case "READ_ACP":
+			permissionTypes = append(permissionTypes, obs.PermissionReadAcp)
+		case "WRITE_ACP":
+			permissionTypes = append(permissionTypes, obs.PermissionWriteAcp)
+		}
+	}
+	return permissionTypes
+}
+
+func appendUserTypeGrants(rawMap map[string]interface{}, grants []obs.Grant, domainID string) []obs.Grant {
+	permissionTypes := getPermissionTypesFromRawMap(rawMap)
+	log.Printf("[DEBUG] The grantee user permission types: %v.", permissionTypes)
+	for _, permissionType := range permissionTypes {
+		grants = append(grants, obs.Grant{
+			Permission: permissionType,
+			Grantee: obs.Grantee{
+				Type: obs.GranteeUser,
+				ID:   domainID,
+			},
+		})
+	}
+	return grants
+}
+
+func appendGroupTypeGrants(rawMap map[string]interface{}, grants []obs.Grant, uriType obs.GroupUriType) []obs.Grant {
+	permissionTypes := getPermissionTypesFromRawMap(rawMap)
+	log.Printf("[DEBUG] The grantee group permission types: %v.", permissionTypes)
+	for _, permissionType := range permissionTypes {
+		grants = append(grants, obs.Grant{
+			Permission: permissionType,
+			Grantee: obs.Grantee{
+				Type: obs.GranteeGroup,
+				URI:  uriType,
+			},
+		})
+	}
+	return grants
+}
+
+func buildObsBucketAclGrantsParam(d *schema.ResourceData, cfg *config.Config) []obs.Grant {
+	var grants []obs.Grant
+	if rawArray, ok := d.Get("owner_permission").([]interface{}); ok && len(rawArray) > 0 {
```

Comment:
what will happen if `owner_permission` is not configured?

Reply:
The owner has changed to computed.
When the owner_permission is not configured, the current obs bucket acl owner permission will be used.